File: 0305/SeoyeonHong/11724 연결 요소의 개수.py
import sys
sys.setrecursionlimit(50000000)
input = sys.stdin.readline
N, M = map(int, input().split())
graph = [[] for _ in range(N+1)]
visited = [False for _ in range(N+1)]
count = 0

for _ in range(M):
    u, v = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)

def dfs(s):
    for v in graph[s]:
        if not visited[v]:
            visited[v] = True
            dfs(v)

for i in range(1, N+1):
    if not visited[i]:
        visited[i] = True
        dfs(i)
        count += 1

print(count)

File: 0305/SoohyunKim/11724_연결요소의개수.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);

let graph = Array.from(Array(N + 1), () => []);
for (let i = 0; i < M; i++) {
  let [from, to] = input[i + 1].split(' ').map(Number);
  graph[from].push(to);
  graph[to].push(from);
}
let visited = Array(N + 1).fill(false);

function dfs(start) {
  visited[start] = true;
  
  for (let i = 0; i < graph[start].length; i++) {
    let next = graph[start][i];
    if (!visited[next]) {
      dfs(next);
    }
  }
}

let ans = 0;
for (let i = 1; i <= N; i++) {
  if (!visited[i]) {
    dfs(i);
    ans++;
  }
}

console.log(ans);